packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -486,19 +499,36 @@ export class VueElement extends BaseClass {
     return vnode
   }
 
-  private _applyStyles(styles: string[] | undefined) {
-    const root = this.shadowRoot
-    if (!root) return
-    if (styles) {
-      styles.forEach(css => {
-        const s = document.createElement('style')
-        s.textContent = css
-        root.appendChild(s)
-        // record for HMR
-        if (__DEV__) {
+  private _applyStyles(
+    styles: string[] | undefined,
+    owner?: ConcreteComponent,
+  ) {
+    if (!styles) return
+    if (owner) {
+      if (owner === this._def || this._styleChildren.has(owner)) {
+        return
+      }
+      this._styleChildren.add(owner)
+    }
+    for (let i = styles.length - 1; i >= 0; i--) {
+      const s = document.createElement('style')
+      s.textContent = styles[i]
+      this.shadowRoot!.prepend(s)
```

Comment:
@yyx990803  Can we do the append? As my global styles are in the last tag and override child styles. In case of append we would  have child styles come after the global. Maybe an option to specify this?